cmd/ztest/ztest.c

```diff
@@ -5397,6 +5400,57 @@ ztest_spa_rename(ztest_ds_t *zd, uint64_t id)
 	(void) rw_unlock(&ztest_name_lock);
 }
 
+void
+ztest_fletcher(ztest_ds_t *zd, uint64_t id)
+{
+	hrtime_t end = gethrtime() + NANOSEC;
+
+	srandom(time(NULL));
+
+	fletcher_4_init();
+	while (gethrtime() <= end) {
+		int run_count = 100;
+		const uint32_t maxsize = 16 * 1024 * 1024;
+		void *buf;
+		uint32_t size;
+		int *ptr;
+		int i;
+		zio_cksum_t zc_ref;
+		zio_cksum_t zc_ref_byteswap;
+
+		/* 512 bytes aligned */
+		size = (random() | 0x10000) % maxsize & ~511;
+		buf = malloc(size);
+		if (buf == NULL) {
+			fputs("out of memory", stderr);
+			return;
+		}
+
+		for (i = 0, ptr = buf; i < size / sizeof (*ptr); i++, ptr++)
+			*ptr = random();
```

Comment:
You can use `ztest_random()` here and drop the `srandom()` initialization above.